pkg/query-service/agentConf/manager.go

```diff
@@ -0,0 +1,217 @@
+package agentConf
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/jmoiron/sqlx"
+	"go.signoz.io/signoz/pkg/query-service/app/opamp"
+	filterprocessor "go.signoz.io/signoz/pkg/query-service/app/opamp/otelconfig/filterprocessor"
+	tsp "go.signoz.io/signoz/pkg/query-service/app/opamp/otelconfig/tailsampler"
+	"go.uber.org/zap"
+	yaml "gopkg.in/yaml.v3"
+)
+
+var m *Manager
+
+func init() {
+	m = &Manager{}
+}
+
+type Manager struct {
+	Repo
+	// lock to make sure only one update is sent to remote agents at a time
+	lock uint32
+}
+
+// Ready indicates if Manager can accept new config update requests
+func (mgr *Manager) Ready() bool {
+	if mgr.lock != 0 {
```

Comment:
This was misssed as i had used mutex before. Will add atomic.LoadUint64

Atomic is used to check if lock is acquired by other process.  Couldnt find a way to check lock in mutex

Reply:
here we are using a singleton pattern to make config requests mutually exclusive. We are blocking users in the UI when a config update is already in progress. This is to reduce concurrency complications.  
I understand the approach is not native to go. but this was decided in earlier calls (if you remember) to avoid concurrency issues